### ISSUE TITLE
💅  `useNamingConvention` can suggest useless fix

### ISSUE THREAD

**Issue by xnuk** [biomejs/biome#1786]
### Environment information
Used [`biome-linux-x64` binary from the release](https://github.com/biomejs/biome/releases/tag/cli%2Fv1.5.3) (sha256sum: `adf8a6029f43ac6eb07c86519f7ff08875915acec082d0be9393888044806243`)

```
CLI:
  Version:                      1.5.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "foot"
  JS_RUNTIME_VERSION:           unset
  JS_RUNTIME_NAME:              unset
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Rule name

[lint/style/useNamingConvention](https://biomejs.dev/linter/rules/use-naming-convention/)

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=ZQB4AHAAbwByAHQAIABjAG8AbgBzAHQAIABkAGEAdABhACAAPQAgAHsAIABIxVWxWNU4wZTGOgAgADEAIAB9AAoA

### Expected result

Currently it shows this:
```typescript
  ⚠ This object property name should be in camelCase.
  
  > 1 │ export const data = { 안녕하세요: 1 }
      │                       ^^^^^^^^^^
    2 │ 
  
  ℹ The name could be renamed to `안녕하세요`.
```

Since the suggested fix is same as original, the message should be removed or replaced to something else. Disabling the lint could be an option.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
I suppose we should ignore identifiers that aren't UTF-8

**Comment by nhedger** [comment]
> I suppose we should ignore identifiers that aren't UTF-8

AFAIK, Korean characters can be represented as UTF-8.

Not 100% sure, but most likely concepts such as camel casing and such only apply to _latin_ scripts.

**Comment by xnuk** [comment]
Note that Koreans mostly do not prefer to write variable names in Korean, except when they have to communicate with outside services, written by who prefers to do (mostly government ones).

**Comment by nhedger** [comment]
@ematipico I was thinking we could change the return type of `Case::identify` from `Case` to `Option<Case>`. This would allow us to represent the fact that there is no case for given identifier. How does that sound?

**Comment by ematipico** [comment]
I'm not very familiar with the rule unfortunately

**Comment by Conaclos** [comment]
> I was thinking we could change the return type of Case::identify from Case to Option<Case>. This would allow us to represent the fact that there is no case for the given identifier. How does that sound?

`Case` has already an `Unknown` variant.
`Case::identify` returns `Case::idemtify` in two cases:

1. when the string mix several cases, e.g `UPPER_lower`
2. when the string uses characters that cannot be in uppercase/lowercase and are not `-` or `_`.

I am not sure how we should handle this.
One idea: we could treat letters that cannot be in lowercase and uppercase as being both in lowercase and uppercase. This will require adding a new variant `Case::LowerUpper`.

**Comment by nhedger** [comment]
My reasoning was that these scripts _might not have a case at all_, so giving them a case variant of `LowerUpper` seems odd.

**Comment by Conaclos** [comment]
> My reasoning was that these scripts _might not have a case at all_, so giving them a case variant of `LowerUpper,` or any case for that matter seems odd.

I didn't check the implantation of the eslint rule, however the rule seems to treat these characters as both upper and lower cases.

Otherwise, what do you suggest? What about mixing latin characters with such characters?
